docs/development/developer_guide.md

```diff
@@ -128,24 +144,93 @@ kind load docker-image my-username/training-operator:my-pr-01
 cd ./manifests/overlays/standalone
 kustomize edit set image my-username/training-operator=my-username/training-operator:my-pr-01
 ```
+
 Update the `newTag` key in `./manifests/overlayes/standalone/kustimization.yaml` with the new image.
 
 Deploy the operator with:
+
 ```sh
 kubectl apply -k ./manifests/overlays/standalone
 ```
+
 And now we can submit jobs to the operator.
+
 ```sh
 kubectl patch -n kubeflow deployments training-operator --type json -p '[{"op": "replace", "path": "/spec/template/spec/containers/0/image", "value": "my-username/training-operator:my-pr-01"}]'
 kubectl apply -f https://raw.githubusercontent.com/kubeflow/training-operator/master/examples/pytorch/simple.yaml
 ```
+
 You should be able to see a pod for your training operator running in your namespace using
+
 ```
 kubectl logs -n kubeflow -l training.kubeflow.org/job-name=pytorch-simple
 ```
+
+## Testing changes locally without build image
+
+Building and testing changes through container images can be time-consuming, so here is a simpler method that allows you
+to start and test directly through the command line or your development tools. Note that this approach is effective only
+for clusters created with Kind on your local machine (e.g., on a Mac).
+
+### Install cert-manager and generate a certificate
```

Comment:
I don't think it's necessary to use cert-manager; the cert-controller will create the certificates.

Reply:
Yes, we don't need cert-manger since cert generation is part of the manager: https://github.com/kubeflow/training-operator/blob/master/cmd/training-operator.v1/main.go#L156C16-L159